virtualcluster/cmd/manager/main.go

```diff
@@ -67,8 +69,12 @@ func main() {
 	flag.StringVar(&controlPlaneProvisioner, "provisioner", "native",
 		"The underlying platform that will provision control plane for virtualcluster.")
 	flag.BoolVar(&leaderElection, "leader-election", true, "If enable leaderelection for vc-manager")
-	flag.StringVar(&leaderElectionCmName, "le-cm-name", "vc-manager-leaderelection-lock",
-		"The name of the configmap that will be used as the resourcelook for leaderelection")
+	// Deprecated: the flag used resource type as part of the name. Replaced by le-name.
+	flag.StringVar(&leaderElectionCmName, "le-cm-name", "", "DEPRECATED. Use --le-name instead")
+	flag.StringVar(&leaderElectionID, "le-name", "vc-manager-leaderelection-lock",
+		"The name of the resource that will be used as the resourcelock for leaderelection")
+	flag.StringVar(&leaderElectionResourceLock, "le-resource", "configmapsleases",
+		"The type of the resource that will be used as the resourcelock for leaderelection [configmaps, leases, configmapsleases]")
```

Comment:
Can we change these to line up with core kubernetes leader election flag names - https://github.com/kubernetes/component-base/blob/25833dc857cd1d3dd4ae46014ada3392729729bf/config/options/leaderelectionconfig.go#L25-L53